packages/@tinacms/cli/src/cmds/init/codegen/index.ts

```diff
@@ -0,0 +1,403 @@
+    }
+    return visit
+  }
+
+/**
+ * Creates a TypeScript transformation visitor that updates an object literal property of a given function call.
+ *
+ * @param {ts.SourceFile} sourceFile - The source file containing the function call.
+ * @param {string} functionName - The name of the function to update.
+ * @param {string} propertyName - The name of the property to update.
+ * @param {ts.SourceFile} propertyValueExpressionSourceFile - The source file containing the expression for the new property value.
+ * @param {ts.Expression} propertyValue - The new property value.
+ *
+ * @returns {ts.TransformerFactory<ts.SourceFile>} The transformation visitor.
+ */
+const makeUpdateObjectLiteralPropertyVisitor =
+  (
+    sourceFile: ts.SourceFile,
+    functionName: string,
+    propertyName: string,
+    propertyValueExpressionSourceFile: ts.SourceFile,
+    propertyValue: ts.Expression
+  ): ts.TransformerFactory<ts.SourceFile> =>
+  (ctx: ts.TransformationContext) => {
+    const visitor = (node: ts.Node) => {
+      if (
+        ts.isCallExpression(node) &&
+        ts.isIdentifier(node.expression) &&
+        node.expression.text === functionName &&
+        node.arguments.length > 0 &&
+        ts.isObjectLiteralExpression(node.arguments[0])
+      ) {
+        let foundProperty = false
+        const configObject = node.arguments[0] as ts.ObjectLiteralExpression
+        // Map the properties of the first argument
+        const updateProperties = configObject.properties.map((property) => {
+          if (
+            ts.isPropertyAssignment(property) ||
+            ts.isShorthandPropertyAssignment(property)
+          ) {
+            const name = property.name.getText(sourceFile)
+            if (name === propertyName) {
+              foundProperty = true
+              return ts.factory.createPropertyAssignment(name, propertyValue)
+            }
+          }
+          return property
+        })
+
+        // If the property wasn't found, add it
+        if (!foundProperty) {
+          updateProperties.unshift(
+            ts.factory.createPropertyAssignment(propertyName, propertyValue)
+          )
+        }
+
+        return ts.factory.createCallExpression(
+          node.expression,
+          node.typeArguments,
+          [ts.factory.createObjectLiteralExpression(updateProperties, true)]
+        ) as ts.Node
+      }
+
+      return ts.visitEachChild(node, visitor, ctx)
+    }
+    return visitor
+  }
+
+export const addSelfHostedTinaAuthToConfig = async (
+  config: Config,
+  configFile: GeneratedFile
+) => {
+  const pathToConfig = configFile.resolve(config.typescript).path
+  const sourceFile = ts.createSourceFile(
+    pathToConfig,
+    fs.readFileSync(pathToConfig, 'utf8'),
+    config.typescript ? ts.ScriptTarget.Latest : ts.ScriptTarget.ESNext
+  )
+
+  const transformedSourceFileResult = ts.transform(
+    sourceFile,
+    [
+      makeImportsVisitor(sourceFile, {
```

Comment:
You _could_ get these values from ` config.authenticationProvider.configImports` instead of hard coding them.